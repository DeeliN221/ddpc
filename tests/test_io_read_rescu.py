"""Tests for the read functions in ddpc.io.read.dspaw_as."""

from pathlib import Path

from ddpc.io.read import rescu_xyz
from loguru import logger


def test_read(snapshot):
    """Tests reading various as files."""
    for s in sorted(Path(f"{__file__}/../structures").resolve().iterdir()):
        if s.name.endswith(".xyz"):
            atoms = rescu_xyz.read(s)
            logger.info(atoms)
            assert f"{s.name}\n{atoms}" == snapshot
