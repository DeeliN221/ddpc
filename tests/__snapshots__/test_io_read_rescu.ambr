# serializer version: 1
# name: test_read
  '''
  BN.xyz
  Atoms(symbols='NB', pbc=True)
  '''
# ---
# name: test_read.1
  '''
  Si.xyz
  Atoms(symbols='Si2', pbc=True, initial_magmoms=...)
  '''
# ---
# name: test_read.2
  '''
  SiTi.xyz
  Atoms(symbols='Si254Ti54', pbc=True)
  '''
# ---
# name: test_read.3
  '''
  graphene.xyz
  Atoms(symbols='C2', pbc=True)
  '''
# ---
# name: test_read.4
  '''
  huge.xyz
  Atoms(symbols='C9842B9842N9842', pbc=True)
  '''
# ---
